docs/protocol/Constants.md

```diff
@@ -48,7 +37,6 @@ Coins exist over a network and have a distinct integer ID.
 | Coin     | Network  | ID |
 |----------|----------|----|
 | Bitcoin  | Bitcoin  | 0  |
-| Ethereum | Ethereum | 1  |
-| USDC     | Ethereum | 2  |
-| DAI      | Ethereum | 3  |
-| Monero   | Monero   | 4  |
+| Ether    | Ethereum | 1  |
+| DAI      | Ethereum | 2  |
```

Comment:
Perhaps CAIP-19 specification can be used: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md. It describes how to identify assets in multi-chain applications.

Reply:
Absolute NACK around GUIDs. Alternatively to an integer, a string could be used. My objections are the inherent inefficiencies and the fact it's a protocol acknowledgement of what the coins are, rather than just another coin.

These inefficiencies aren't just minute. Bitcoin instructions are almost at the OP_RET size limit as-is. If integrators still are forced to a number scheme, I don't care to try to offer a nicety sporadically. I'd rather just have our lib define the constants as expected. And then for a native Substrate TX, I think this would bloat them 10%? As it adds ~12 bytes to each TX.

Alternatively, we can have a numeric ID for reference *AND* an on-chain string label, which is called upon request, not causing any inefficiencies. This on-chain label would be explicitly formatted so clients could get asset metadata without issue. CAIP-19 seems generally competent (URI scheme, chain ID, class, ID). My gripes are reuse of `:` (instead of a further `/`) and usage of SLIP-44 instead of a magic like `self` for the base.

So I do think the best move is an additional metadata section. My immediate question is should that be here, a new pallet entirely, and do we use CAIP-19 as-is, or a more LibP2P/Substrate esque `eip155:1/native`, `eip155:1/erc20:0x6b175474e89094c44da98b954eedeac495271d0f`...

Reply:
Note that Monero currently doesn't have a dedicated chain ID in CAIP system.
I've recently created an issue for that: https://github.com/ChainAgnostic/namespaces/issues/41 (feedback is appreciated!).

Reply:
I prefer to use string to integers from a security perspective. As far as CAIP goes, the effort is indeed appreciated but I generally feel we're a bit early in the cross-chain pursuit to define fully backwards compatible standards that will account for the needs of various environments. I believe at one point we discussed defining our own steganography towards efficiency of data included on other chains. I'm a bit of an optimization maximalist and hope we can even be compatible if chains more restrictive of metadata arise.

Maybe we should keep this a simplistic structure for protonet, keep CAIP in mind for testnet+, and focus this discussion on implementing a distinction between tokens and native assets?